packages/integration-tests/suites/replay/privacyInput/test.ts

```diff
@@ -0,0 +1,95 @@
+import { expect } from '@playwright/test';
+import { IncrementalSource } from '@sentry-internal/rrweb';
+import type { incrementalData, inputData} from '@sentry-internal/rrweb/typings/types';
+
+import { sentryTest } from '../../../utils/fixtures';
+import {
+  getIncrementalRecordingSnapshots,
+  shouldSkipReplayTest,
+  waitForReplayRequest,
+} from '../../../utils/replayHelpers';
+
+function isInputMutation(snap: incrementalData): snap is inputData {
+  return snap.source == IncrementalSource.Input;
+}
+
+sentryTest('should mask input initial value and its changes', async ({ getLocalTestPath, page }) => {
+  if (shouldSkipReplayTest()) {
+    sentryTest.skip();
+  }
+
+  const reqPromise0 = waitForReplayRequest(page, 0);
+  const reqPromise1 = waitForReplayRequest(page, 1);
+  const reqPromise2 = waitForReplayRequest(page, 2);
+  const reqPromise3 = waitForReplayRequest(page, 3);
+
+  await page.route('https://dsn.ingest.sentry.io/**/*', route => {
+    return route.fulfill({
+      status: 200,
+      contentType: 'application/json',
+      body: JSON.stringify({ id: 'test-id' }),
+    });
+  });
+
+  const url = await getLocalTestPath({ testDir: __dirname });
+
+  await page.goto(url);
+
+  await reqPromise0;
+
+  const text = 'this is test text';
+
+  await page.locator('#input').type(text)
+  const snapshots = getIncrementalRecordingSnapshots(await reqPromise1).filter(isInputMutation);
+  const lastSnapshot = snapshots[snapshots.length-1];
+  expect(lastSnapshot.text).toBe(text);
```

Comment:
Can we be sure that this snapshot always comes last? If yes, no problem! Just asking because I observed quite some flakiness with incremental snapshots across different browsers.

Reply:
Yeah I observed a bunch of flake too from mouse movements I think -- having the `isInputMutation` filter helps with that and those are sorted (otherwise playing back input changes would not work as expected).